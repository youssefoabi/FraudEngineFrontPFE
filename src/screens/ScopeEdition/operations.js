import { call, put, takeLatest } from 'redux-saga/effects';
import { prop } from 'ramda';

import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import {
  SAVE_EDIT_SCOPE_ERROR,
  SAVE_EDIT_SCOPE_REQUEST,
  SAVE_EDIT_SCOPE_SUCCESS,
} from '../../components/ScopeForm/actions';
import clientApi from '../../client-api';

export function* saveScope({ payload }) {
  const { scope } = payload;
  if (!scope) return null;
  yield put({ type: PENDING_REQUEST });

  try {
    yield call(clientApi().saveEditScope, { ...scope, scopeName: prop('name', scope) });

    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: SAVE_EDIT_SCOPE_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: SAVE_EDIT_SCOPE_ERROR,
      error: { errorMessage: "Erreur d'enregistrement de la gamme" },
    });
  }

  return true;
}

export default [takeLatest(SAVE_EDIT_SCOPE_REQUEST, saveScope)];
