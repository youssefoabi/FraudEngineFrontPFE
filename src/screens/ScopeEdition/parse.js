import { isNil } from 'ramda';

const toTimestamp = date => (date ? Number(new Date(date)) : null);

const parseScope = scope => {
  if (isNil(scope)) return null;

  const { id, modifiedAt, name, urlTemplate } = scope;

  return {
    id,
    modifiedAt: toTimestamp(modifiedAt),
    name,
    urlTemplate,
  };
};

export default parseScope;
