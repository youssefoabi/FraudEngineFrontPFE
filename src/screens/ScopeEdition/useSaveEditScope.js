import { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { SAVE_EDIT_SCOPE_REQUEST } from '../../components/ScopeForm/actions';

export default function useSaveEditScope() {
  const dispatch = useDispatch();

  const saveEditScope = useCallback(
    scope => dispatch({ type: SAVE_EDIT_SCOPE_REQUEST, payload: { scope } }),
    [dispatch],
  );

  return { saveEditScope };
}
