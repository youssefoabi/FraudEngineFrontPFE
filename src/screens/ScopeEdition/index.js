import { Button, Footer, Screen } from '@cdiscount/backoffice-ui';
import { pathOr, prop } from 'ramda';
import { pathOrNull } from '@cdiscount/ramda-utils';
import { Save } from '@material-ui/icons';
import React, { useEffect, useState } from 'react';
import useReactRouter from 'use-react-router';

import clientApi from '../../client-api';
import parseScope from './parse';
import ScopeForm from '../../components/ScopeForm';
import ServerActionBackdrop from '../../components/ServerActionBackdrop';
import TitleBar from '../../components/TitleBar';
import useSaveEditScope from './useSaveEditScope';

function ScopeEdition() {
  const [scope, setScope] = useState();
  const { match } = useReactRouter();

  const { saveEditScope } = useSaveEditScope();

  useEffect(() => {
    const id = pathOrNull(['params', 'id'], match);
    clientApi()
      .getScope(id)
      .then(prop('data'))
      .then(rawScope => {
        setScope(parseScope(rawScope));
      });
  }, [match]);

  const name = pathOr('', ['name'], scope);

  return (
    <Screen
      titlebar={
        <TitleBar
          title={`Edition de la gamme : ${name}`}
          actions={[
            {
              id: 'save_scope_button',
              label: 'Enregistrer',
              icon: Save,
              size: 'large',
              onClick: () => saveEditScope(scope),
            },
          ]}
        />
      }
      bottombar={
        <Footer>
          <Button
            onClick={() => saveEditScope(scope)}
            startIcon={<Save />}
            size="large"
            color="tertiary"
            label="Enregistrer"
            variant="contained"
          />
        </Footer>
      }
    >
      <ScopeForm scope={scope} setScope={setScope} />
      <ServerActionBackdrop />
    </Screen>
  );
}

export default ScopeEdition;
