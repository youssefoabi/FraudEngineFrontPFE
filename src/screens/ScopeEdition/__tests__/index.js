import { itRendersAllMutations } from '@cdiscount/jest-utils';

import clientApi from '../../../client-api';
import ScopeEdition from '../';

jest.mock('../../../client-api');

beforeEach(() => {
  clientApi.mockClear();
});

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
}));

jest.mock('use-react-router', () => () => ({ match: {} }));

describe(ScopeEdition.name, () => {
  itRendersAllMutations(ScopeEdition, [{ name: 'without props', props: {} }]);
});
