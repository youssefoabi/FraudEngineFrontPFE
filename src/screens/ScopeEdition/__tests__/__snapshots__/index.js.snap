// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopeEdition ScopeEdition renders correctly without props 1`] = `
<o
  bottombar={
    <p
      backToTopLabel="Haut de page"
    >
      <y
        color="tertiary"
        icon={null}
        label="Enregistrer"
        onClick={[Function]}
        size="large"
        startIcon={<UNDEFINED />}
        variant="contained"
      />
    </p>
  }
  titlebar={
    <withRouter(TitleBar)
      actions={
        Array [
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "SaveIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": "save_scope_button",
            "label": "Enregistrer",
            "onClick": [Function],
            "size": "large",
          },
        ]
      }
      title="Edition de la gamme : "
    />
  }
>
  <ScopeForm
    scope={null}
    setScope={[Function]}
  />
  <ServerActionBackdrop />
</o>
`;
